e2e-tests/desktop/tests/credentials.spec.js

```diff
@@ -0,0 +1,285 @@
+    const clientToken = vaultToken.auth.client_token;
+
+    // Create a vault credential store and generic secrets type credential library
+    const vaultCredentialStore = await boundaryHttp.createVaultCredentialStore(
+      request,
+      project.id,
+      vaultAddr,
+      clientToken,
+    );
+    credentialLibrary = await boundaryHttp.createVaultCredentialGenericSecret(
+      request,
+      {
+        credentialStoreId: vaultCredentialStore.id,
+        vaultPath: `${secretsPath}/data/${secretName}`,
+      },
+    );
+
+    // Create a tcp target with vault brokered credential Libraries
+    targetWithBrokeredVaultCredentials = await boundaryHttp.createTarget(
+      request,
+      {
+        scopeId: project.id,
+        type: 'tcp',
+        port: targetPort,
+      },
+    );
+    targetWithBrokeredVaultCredentials = await boundaryHttp.addHostSource(
+      request,
+      {
+        target: targetWithBrokeredVaultCredentials,
+        hostSourceIds: [hostSet.id],
+      },
+    );
+    targetWithBrokeredVaultCredentials =
+      await boundaryHttp.addBrokeredCredentials(request, {
+        target: targetWithBrokeredVaultCredentials,
+        credentialIds: [credentialLibrary.id],
+      });
+
+    // Create a tcp target with static brokered credentials
+    targetWithBrokeredStaticCredentials = await boundaryHttp.createTarget(
+      request,
+      {
+        scopeId: project.id,
+        type: 'tcp',
+        port: targetPort,
+      },
+    );
+    targetWithBrokeredStaticCredentials = await boundaryHttp.addHostSource(
+      request,
+      {
+        target: targetWithBrokeredStaticCredentials,
+        hostSourceIds: [hostSet.id],
+      },
+    );
+    targetWithBrokeredStaticCredentials =
+      await boundaryHttp.addBrokeredCredentials(request, {
+        target: targetWithBrokeredStaticCredentials,
+        credentialIds: [sshPrivateKeyCredential.id, jsonCredential.id],
+      });
+  },
+);
+
+test.afterEach(async ({ request }) => {
+  if (org) {
+    await boundaryHttp.deleteOrg(request, org.id);
+  }
+});
+
+test.describe('Credential tests', async () => {
+  test('Vault brokered Credentials are displayed as expected', async ({
+    authedPage,
+  }) => {
+    await authedPage
+      .getByRole('link', { name: targetWithBrokeredVaultCredentials.name })
+      .click();
+    await authedPage.getByRole('button', { name: 'Connect' }).click();
+    await expect(
+      authedPage.getByRole('heading', { name: 'Sessions' }),
+    ).toBeVisible();
+
+    await expect(authedPage.getByText(credentialLibrary.name)).toBeVisible();
+    await expect(authedPage.getByText('password')).toBeVisible();
+
+    await authedPage
+      .getByRole('listitem')
+      .filter({ hasText: 'password' })
+      .getByLabel('Toggle secret visibility')
+      .click();
+    await expect(
+      authedPage
+        .getByRole('listitem')
+        .filter({ hasText: 'password' })
+        .locator('pre'),
+    ).toHaveText(String.raw`pass\word`);
```

Comment:
How come this is ``String.raw`pass\word` ``  since on [line 93](https://github.com/hashicorp/boundary-ui/pull/2660/files#diff-411688c020c9741d39b6c48029c502b7b4c831aa55f11a0f40dbfbe496bf7c77R93) it's ``String.raw`pass\\word` ``?

Reply:
without escaping the `\`, using single quotes around the password like '`String.raw`pass\word`'  seems to work fine when passed to vault but eventually it times out and crashes, so I tried escaping the backslash.

Reply:
I see, so this is how vault handles escape characters?

Reply:
so I tested this using spawnSync and it didn't need an esc character, I'm thinking some weird issues with special characters in shell env and nothing to do with vault.